### PR TITLE
 Notify the CDC Client if streaming stopped due to exception

### PR DESCRIPTION
**Pull request by azexcy**
Related #22500.

Changes proposed in this pull request:
  - When streaming job failed, send error result to CDC Client
  - Add tinyint decode at MppdbDecodingPlugin

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by azexcy** [comment]
refer https://docs.opengauss.org/en/docs/3.0.0/docs/BriefTutorial/data-types.html tinyint

**Comment by azexcy** [comment]
image

File: kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java [hunk 1]
```diff
@@ -200,6 +202,10 @@ public void onFailure(final Throwable throwable) {
             log.error("onFailure, {} task execute failed.", identifier, throwable);
             String jobId = jobItemContext.getJobId();
             jobAPI.updateJobItemErrorMessage(jobId, jobItemContext.getShardingItem(), throwable);
+            if (jobItemContext.getSink() instanceof CDCSocketSink) {
+                CDCSocketSink cdcSink = (CDCSocketSink) jobItemContext.getSink();
+                cdcSink.getChannel().writeAndFlush(CDCResponseUtils.failed("", "", throwable.getMessage()));
```
Comment (sandynz) [inline kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java:207]: Looks channel.writeAndFlush might be invoked in multi-thread here (CDCSocketSink is still running), could we verify the thread-safety of channel write?

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java (hunk 1 above)
Comment (azexcy) [reply kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java:207]: I find some related question: 
https://lists.jboss.org/pipermail/netty-users/2008-November/000045.html
https://stackoverflow.com/questions/23161608/is-netty4-channelhandlercontext-write-thread-safe

Comment (sandynz) [reply kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java:207]: OK

**Review by sandynz — APPROVED** [review]

**Pull request merged**